lib/api/route/channels.dart

```diff
@@ -4,6 +4,42 @@ import '../core.dart';
 import '../model/model.dart';
 part 'channels.g.dart';
 
+/// https://zulip.com/api/subscribe
+///
+/// [subscriptions] is a list of channel names.
+/// (This is one of the few remaining areas where the Zulip API hasn't migrated
+/// to using IDs.)
+Future<void> subscribeToChannel(ApiConnection connection, {
+  // TODO(server-future): This should use a stream ID, not stream name.
+  //   (Keep dartdoc up to date.)
+  //   Server issue: https://github.com/zulip/zulip/issues/10744
+  required List<String> subscriptions,
+  List<int>? principals,
+}) {
+  return connection.post('subscribeToChannel', (_) {}, 'users/me/subscriptions', {
+    'subscriptions': subscriptions.map((subName) => {'name': subName}).toList(),
```

Comment:
nit: the name "name" seems clear enough, and avoids making up a distinct name for the same thing:
```suggestion
    'subscriptions': subscriptions.map((name) => {'name': name}).toList(),
```